fluidkeys/main.go

```diff
@@ -254,24 +259,57 @@ func printSecretKeyListing(listNumber int, key gpgwrapper.SecretKeyListing) stri
 	return output
 }
 
-func promptForKeyToImportFromGpg(secretKeyListings []gpgwrapper.SecretKeyListing) gpgwrapper.SecretKeyListing {
+func promptForKeyToImportFromGpg(secretKeyListings []gpgwrapper.SecretKeyListing) *gpgwrapper.SecretKeyListing {
 	var selectedKey int
-	invalidEntry := fmt.Sprintf("Please select between 1 and %v.\n", len(secretKeyListings))
-	for validInput := false; !validInput; {
-		rangePrompt := colour.LightBlue(fmt.Sprintf("[1-%v]", len(secretKeyListings)))
-		input := promptForInput(fmt.Sprintf(PromptWhichKeyFromGPG + " " + rangePrompt + " "))
-		if integerSelected, err := strconv.Atoi(input); err != nil {
-			fmt.Print(invalidEntry)
+	if len(secretKeyListings) == 1 {
+		onlyKey := secretKeyListings[0]
+		if promptToConfirmImportKeyFromGpg(onlyKey) {
+			return &onlyKey
 		} else {
-			if (integerSelected >= 1) && (integerSelected <= len(secretKeyListings)) {
-				selectedKey = integerSelected - 1
-				validInput = true
-			} else {
+			return nil
+		}
+	} else {
+		invalidEntry := fmt.Sprintf("Please select between 1 and %v.\n", len(secretKeyListings))
+		for validInput := false; !validInput; {
+			rangePrompt := colour.LightBlue(fmt.Sprintf("[1-%v]", len(secretKeyListings)))
+			input := promptForInput(fmt.Sprintf(PromptWhichKeyFromGPG + " " + rangePrompt + " "))
+			if integerSelected, err := strconv.Atoi(input); err != nil {
 				fmt.Print(invalidEntry)
+			} else {
+				if (integerSelected >= 1) && (integerSelected <= len(secretKeyListings)) {
+					selectedKey = integerSelected - 1
+					validInput = true
+				} else {
+					fmt.Print(invalidEntry)
+				}
 			}
 		}
+		return &secretKeyListings[selectedKey]
+	}
+}
+
+func promptToConfirmImportKeyFromGpg(key gpgwrapper.SecretKeyListing) bool {
+	var confirmed bool
+	for validInput := false; !validInput; {
+		input := promptForInput("Import key? [Y/n] ")
+		switch input {
+		case "Y":
```

Comment:
Any reason why not `switch strings.Lower(input)`?

Reply:
👍